homeassistant/components/combined_energy/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow for Combined Energy integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from combined_energy import CombinedEnergy
+from combined_energy.exceptions import (
+    CombinedEnergyAuthError,
+    CombinedEnergyError,
+    CombinedEnergyPermissionError,
+    CombinedEnergyTimeoutError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_INSTALLATION_ID, DEFAULT_NAME, DOMAIN, LOGGER
+
+
+class CombinedEnergyConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Combined Energy."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self._errors: dict[str, str] = {}
+
+    async def _check_installation(
+        self, username: str, password: str, installation_id: int
+    ) -> bool:
+        """Check if we can connect to the combined energy service."""
+        api = CombinedEnergy(
+            username,
+            password,
+            installation_id,
+            session=async_get_clientsession(self.hass),
+        )
+        try:
+            await api.installation()
+        except CombinedEnergyAuthError:
+            self._errors["base"] = "invalid_auth"
+        except CombinedEnergyPermissionError:
+            self._errors[CONF_INSTALLATION_ID] = "installation_not_accessible"
+        except CombinedEnergyTimeoutError:
+            self._errors["base"] = "cannot_connect"
+        except CombinedEnergyError:
+            LOGGER.exception("Unexpected error verifying connection to API")
+            self._errors["base"] = "cannot_connect"
```

Comment:
Is this supposed to be unknown?

Reply:
Yes this should probably have a different error message to indicate to the end user.